pkg/server/server.go

```diff
@@ -200,14 +200,17 @@ func (s *ProxyServer) addBackend(agentID string, conn agent.AgentService_Connect
 			for _, ipv4 := range agentIdentifiers.IPv4 {
 				klog.V(5).InfoS("Add the agent to DestHostBackendManager", "agent address", ipv4)
 				s.BackendManagers[i].AddBackend(ipv4, pkgagent.IPv4, conn)
+				metrics.Metrics.BackendCountInc("DestHostBackendManager", "ipv4")
```

Comment:
I think it's leaving this logic in `AddBackend`  is preferable, which makes adding backend logic more clear.
How about adding filed to backend storage to differentiate backend count for different kinds of backend managers.

Reply:
Please review the code.